services/certmanager/resources/client.go

```diff
@@ -0,0 +1,53 @@
+package resources
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ionos-cloud/ionosctl/pkg/config"
+	sdkgo "github.com/ionos-cloud/sdk-go-cert-manager"
+	"github.com/spf13/viper"
+)
+
+type Client struct {
+	sdkgo.APIClient
+}
+
+type ClientConfig struct {
+	sdkgo.Configuration
+}
+
+// ClientService is a wrapper around ionoscloud.APIClient
+type ClientService interface {
+	Get() *Client
+	GetConfig() *ClientConfig
+}
+
+type clientService struct {
+	client *sdkgo.APIClient
+}
+
+var _ ClientService = &clientService{}
+
+func NewClientService(name, pwd, token, hostUrl string) (ClientService, error) {
+	if token == "" && (name == "" || pwd == "") {
```

Comment:
I think if the err check is moved outside the init chain, you wouldn't need to throw an error inside the constructor. To remember as a refactor at some point. Also the err message can be improved to mention that you need to provide either username and pass, or token.

Reply:
@cristiGuranIonos again, fully agree with this.... buuuut, this is the case for every single service. Same discussion as above

Duplicated code:

https://github.com/ionos-cloud/ionosctl/blob/master/services/cloudapi-v6/resources/client.go#L34-L45

https://github.com/ionos-cloud/ionosctl/blob/master/services/dbaas-postgres/resources/client.go#L32-L41

Again, maybe handle this in a different PR for consistency sake?

Reply:
I saw, yes definitely in another PR :)